### PR TITLE
Feature Constitution: Previous  Info

### PR DESCRIPTION
**Pull request by RicardoJ**

### CODE REVIEW

File: services/integration/src/models/constitution/PreviousInformation.model.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  viability: {
+    type: String,
+    required: [true, 'Viability obligatory field']
+  },
+  requirements: {
+    type: String,
+    required: [true, 'Requirements obligatory field']
+  },
+  client: {
```
Comment (juribee) [inline services/integration/src/models/constitution/PreviousInformation.model.js]: this field must be an oibject with this info: 
image

File: services/integration/src/models/constitution/PreviousInformation.model.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const previousInformationSchema = new Schema({
+  description: {
```
Comment (juribee) [inline services/integration/src/models/constitution/PreviousInformation.model.js]: Este schema necesita un Ptoject id para poder relacionar esta info con un project especifico

File: services/integration/src/routes/constitution/PreviousInformation.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const Router = require('express').Router();
+const previousInformation = require('../../controllers/constitution/previousInformation');
+
+Router.post('/', previousInformation.create);
+Router.get('/', previousInformation.find);
+Router.patch('/:id', previousInformation.update);
```
Comment (juribee) [inline services/integration/src/routes/constitution/PreviousInformation.js]: Ya que un proyecto solo tiene un solo bloque de `prev Info` entonces no es necesario agregar otro id para actualizar esta info. Bastan con saber el id del proyecto.

File: services/integration/src/controllers/constitution/PreviousInformation.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+const find = asyncHandler(async (req, res) => {
+  const previousInformation = await PreviousInformation.find({});
+
+  res.send(previousInformation);
+});
+
+const update = asyncHandler(async (req, res) => {
+  const { id } = req.params;
+  const updatedDocument = await PreviousInformation.findByIdAndUpdate(id, {
```
Comment (juribee) [inline services/integration/src/controllers/constitution/PreviousInformation.js]: Here all you need is the project id to be able find the prev-info

File: services/integration/src/models/constitution/PreviousInformation.model.js (hunk 1 above)
Comment (RicardoJ) [reply services/integration/src/models/constitution/PreviousInformation.model.js]: ok , but in trello you dont say this

File: services/integration/src/models/constitution/PreviousInformation.model.js (hunk 2 above)
Comment (RicardoJ) [reply services/integration/src/models/constitution/PreviousInformation.model.js]: really? , yesterday i asked you , and you told me that is not necessary

File: services/integration/src/routes/constitution/PreviousInformation.js (hunk 3 above)
Comment (RicardoJ) [reply services/integration/src/routes/constitution/PreviousInformation.js]: entonces la ruta para el patch seria simplemente /    ?

File: services/integration/src/models/constitution/PreviousInformation.model.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const previousInformationSchema = new Schema({
+  projectId: {
+    type: String,
+    required: [true, 'Project Id obligatory field'],
```
Comment (juribee) [inline services/integration/src/models/constitution/PreviousInformation.model.js]: ``projectId` is an required field` should be right message here, please use this patter for every required field

File: services/integration/src/controllers/constitution/PreviousInformation.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+const PreviousInformation = require('../../models/constitution/PreviousInformation.model');
```
Comment (juribee) [inline services/integration/src/controllers/constitution/PreviousInformation.js]: This directory should be named `previousInformation` or even shorter `previousInfo`

Comment (RicardoJ) [reply services/integration/src/controllers/constitution/PreviousInformation.js]: i did changes

File: services/integration/src/models/constitution/PreviousInformation.model.js (hunk 5 above)
Comment (RicardoJ) [reply services/integration/src/models/constitution/PreviousInformation.model.js]: change is made

**Pull request merged**